app/main/views/suppliers.py

```diff
@@ -512,25 +513,49 @@ def create_new_supplier():
 def duns_number():
     form = DunsNumberForm()
 
-    if request.method == 'POST':
-        if form.validate_on_submit():
-            suppliers = data_api_client.find_suppliers(duns_number=form.duns_number.data)
-            if len(suppliers["suppliers"]) <= 0:
-                session[form.duns_number.name] = form.duns_number.data
-                return redirect(url_for(".company_details"))
+    if request.method == 'GET':
+        if request.args.get('retry', None) == 'true':
+            # When redirected back from the lookup page
+            form.duns_number.errors = ["Enter a different DUNS number"]
+        elif form.duns_number.name in session:
```

Comment:
Thinking out loud here, but is it worth making this just an if, so that the user gets shown the existing duns number they entered even if they get the Enter a different DUNS number? So they can see any obvious errors they made or if they have multiple possible DUNS numbers to try they can see which one to not try again or... iunno.